src/arch/riscv64/CodeGen.zig

```diff
@@ -8441,6 +8441,10 @@ fn failSymbol(func: *Func, comptime format: []const u8, args: anytype) InnerErro
 }
 
 fn parseRegName(name: []const u8) ?Register {
+    // The `fp` alias for `s0` is awkward to fit into the current `Register` scheme, so for now we
+    // special-case it here.
+    if (std.mem.eql(u8, name, "fp")) return .s0;
+
     return std.meta.stringToEnum(Register, name);
```

Comment:
FYI @Rexicon226

Reply:
Seems reasonable to me, maybe we could have a StaticStringMap for the aliases instead?

Reply:
Hmm, that’s a fair point. I think this solution is fine for now, it’s easy to expand later.